### PR TITLE
Fixes 176, Packages are not ignored when using --onlyAllow and --excludePackages

### PR DESCRIPTION
**Pull request by jonny-improbable**
Moves the `--onlyAllow` and `--failOn` checks so that they are performed after the filtering performed by `--packages` and `--excludePackages`.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -368,6 +348,11 @@ exports.init = function(options, callback) {
                 }
             }
         });
+
+        if (!Object.keys(sorted).length) {
```
Comment (jonny-improbable) [inline lib/index.js]: Unrelated drive-by, moved this up to sit after sorting and before an license exclusion / filtering.

**Comment by davglass** [comment]
Published in `license-checker@24.0.0`
